### PR TITLE
docs: add contributing notes about pyca downstream testing

### PR DESCRIPTION
**Pull request by mattsb42-aws**
_Description of changes:_

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: contributing/python/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+The frozen test dependencies are stored in
+`test/upstream-requirements-py27.txt`
+and `test/upstream-requirements-py37.txt`.
+You can update the frozen test dependencies with the
+`freeze-upstream-requirements-py27`
+and `freeze-upstream-requirements-py37`
+`tox` environments.
+You can test the frozen test dependencies with the
+`test-upstream-requirements-py27`
+and `test-upstream-requirements-py27`
```
Comment (farleyb-amazon) [inline contributing/python/README.md:39]: Missed copy/paste -- `test-upstream-requirements-py37`

File: contributing/python/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+The CI for each project requires these test environments to succeed.
+
+The integration tests for both of these libraries
+require certain environment variables to be set.
+However, the non-integration tests MUST complete successfully
+if those environment variables are not set.
+We test this using the
+[`nocmk`](https://github.com/aws/aws-encryption-sdk-python/blob/97d9468375603a708d6fa9cb6703c3fc174501a6/tox.ini#L59-L69)
+`tox` test environment.
+**It is critical that these test environments unset all environment variables.
```
Comment (MatthewBennington) [inline contributing/python/README.md:50]:
```suggestion
**It is critical that this test environment unsets all environment variables.
```

Nit

File: contributing/python/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+In order to avoid this,
+we introduced frozen test dependencies
+that `pyca`'s downstream tests use.
+
+The frozen test dependencies are stored in
+`test/upstream-requirements-py27.txt`
+and `test/upstream-requirements-py37.txt`.
+You can update the frozen test dependencies with the
+`freeze-upstream-requirements-py27`
+and `freeze-upstream-requirements-py37`
```
Comment (farleyb-amazon) [inline contributing/python/README.md:35]: It looks like updating is a manual action, right? Do you have a one-liner recommendation on when or how often we should think about updating these? My gut sense would be to keep them as up-to-date as possible while still ensuring nothing breaks.

File: contributing/python/README.md (hunk 1 above)
Comment (mattsb42-aws) [reply contributing/python/README.md:39]:
```suggestion
and `test-upstream-requirements-py37`
```

File: contributing/python/README.md (hunk 3 above)
Comment (mattsb42-aws) [reply contributing/python/README.md:35]: added

**Review by farleyb-amazon — APPROVED** [review]

**Pull request merged**